types/decimal.go

```diff
@@ -0,0 +1,324 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+// NOTE: never use new(Dec) or else we will panic unmarshalling into the
+// nil embedded big.Int
+type Dec struct {
+	*big.Int `json:"int"`
+}
+
+// number of decimal places
+const Precision = 10
+
+func precisionInt() *big.Int {
+	return new(big.Int).Exp(big.NewInt(10), big.NewInt(Precision), nil)
```

Comment:
Lets cache this result in an `init` function, and just use `new(big.Int).Set` between calls. #postlaunch incorporate a big int pool for usage just within this file.

Reply:
added something here to reuse for precision - I think a lot of reuse potential in this file just need to do it right, different In operations assign things in weird ways - kind of shitty UX in big's side